azurerm/resource_arm_dns_mx_record.go

```diff
@@ -67,150 +68,140 @@ func resourceArmDnsMxRecord() *schema.Resource {
 	}
 }
 
-func resourceArmDnsMxRecordCreate(d *schema.ResourceData, meta interface{}) error {
-	client := meta.(*ArmClient)
-	rivieraClient := client.rivieraClient
+func resourceArmDnsMxRecordCreateOrUpdate(d *schema.ResourceData, meta interface{}) error {
+	dnsClient := meta.(*ArmClient).dnsClient
 
-	tags := d.Get("tags").(map[string]interface{})
-	expandedTags := expandTags(tags)
-
-	createCommand := &dns.CreateMXRecordSet{
-		Name:              d.Get("name").(string),
-		Location:          "global",
-		ResourceGroupName: d.Get("resource_group_name").(string),
-		ZoneName:          d.Get("zone_name").(string),
-		TTL:               d.Get("ttl").(int),
-		Tags:              *expandedTags,
-	}
-
-	mxRecords, recordErr := expandAzureRmDnsMxRecord(d)
-	if recordErr != nil {
-		return fmt.Errorf("Error Building Azure RM MX Record: %s", recordErr)
-	}
-	createCommand.MXRecords = mxRecords
+	name := d.Get("name").(string)
+	resGroup := d.Get("resource_group_name").(string)
+	zoneName := d.Get("zone_name").(string)
+	ttl := int64(d.Get("ttl").(int))
 
-	createRequest := rivieraClient.NewRequest()
-	createRequest.Command = createCommand
+	tags := d.Get("tags").(map[string]interface{})
+	metadata := expandTags(tags)
 
-	createResponse, err := createRequest.Execute()
-	if err != nil {
-		return fmt.Errorf("Error creating DNS MX Record: %s", err)
-	}
-	if !createResponse.IsSuccessful() {
-		return fmt.Errorf("Error creating DNS MX Record: %s", createResponse.Error)
+	records, err := expandAzureRmDnsMxRecords(d)
+	props := dns.RecordSetProperties{
+		Metadata:  metadata,
+		TTL:       &ttl,
+		MxRecords: &records,
 	}
 
-	readRequest := rivieraClient.NewRequest()
-	readRequest.Command = &dns.GetMXRecordSet{
-		Name:              d.Get("name").(string),
-		ResourceGroupName: d.Get("resource_group_name").(string),
-		ZoneName:          d.Get("zone_name").(string),
+	parameters := dns.RecordSet{
+		Name:                &name,
+		RecordSetProperties: &props,
 	}
 
-	readResponse, err := readRequest.Execute()
+	//last parameter is set to empty to allow updates to records after creation
+	// (per SDK, set it to '*' to prevent updates, all other values are ignored)
+	resp, err := dnsClient.CreateOrUpdate(resGroup, zoneName, name, dns.MX, parameters, "", "")
```

Comment:
what's the penultimate argument for here? in general we'd tend to assign the values to variables when there's multiple default values, so that it's clear what the values are